File: Semestr 6/JavaProjekt/Projekt/src/main/java/com/example/projekt/models/Computer.java
package com.example.projekt.models;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
public class Computer {
    @Id
    @GeneratedValue
    private int id;

    private String name;
    private String cpu;
    private double cpuScore;
    private int cpuMaxclockSpeed;
    private int cpuCores;
    private int ram;
    private int ramSpeed;
    private String os;
    private String gpu;
    private String model;
    private Date timeOfPurchase;
    @Column(nullable = true)
    private Integer employeeId;

    @OneToMany(targetEntity = HardDrive.class, cascade = CascadeType.ALL)
    @JoinColumn(name="ch_fk", referencedColumnName = "id")
    private List<HardDrive> hardDrives;

    @OneToMany(targetEntity = Monitor.class, cascade = CascadeType.ALL)
    @JoinColumn(name="cm_fk", referencedColumnName = "id")
    private List<Monitor> monitors;
}

File: Semestr 6/JavaProjekt/Frotnend/src/app/Models/Computer.ts
import { EmployeePC } from "./Employee";

export interface ComputerOld {
    id: number;
    name: string;
    timeOfPurchase: string;
    status: boolean;
    ip: string | null;
}
export enum Dysk{
    "SSD",
    "HDD"
}

export interface HardDrive {
    id: number;
    name: string;
    size: number;
    type: Dysk;
}

export interface Monitor {
    id: number;
    hardwareID: string;
    name: string | null;
}


export interface Computer {
    id: number;
    name: string;
    cpu: string;
    cpuScore: number;
    cpuMaxclockSpeed: number;
    cpuCores: number;
    ram: number;
    ramSpeed: number;
    gpu: string | null;
    os: string;
    model: string | null;
    timeOfPurchase: string;
    hardDrives: HardDrive[];
    monitors: Monitor[];
}

export interface ComputerCard{
    icon: string;
    title: string;
    info: string;
}

export interface ComputerResponse{
    id: number;
    name: string;
    employeeFullName: string;
    cpu: string;
    cpuScore: number;
    ram: number;
    gpu: string;
    os: string;
    hardDriveType: string;
    workType: string;
}

export interface log{
    id: number,
    date: string,
    msg: string,
    pcId: number
  }

  export interface logSummary{
    month: string,
    count: number
  }